.github/workflows/sycl_post_commit.yml

```diff
@@ -11,12 +11,20 @@ on:
     - .github/workflows/sycl_post_commit.yml
 
 jobs:
+  # This job generates matrix of tests for LLVM Test Suite
+  resolve_matrix:
+    name: Resolve Test Matrix
+    uses: ./.github/workflows/sycl_resolve_test_matrix.yml
+    with:
+      lts_config: "l0_gen9"
   linux_default:
     name: Linux Default
+    needs: resolve_matrix
     uses: ./.github/workflows/sycl_linux_build_and_test.yml
     with:
       build_cache_root: "/__w/llvm"
-      build_artifact_suffix: default
+      build_artifact_suffix: "post_commit"
+      lts_matrix: ${{ needs.resolve_matrix.outputs.lts_matrix }}
```

Comment:
Idea for future improvements. Nightly job duplicates a lot of work we do in the post-commit (build + lit + llvm-test-suite), which we can avoid. The first step here is to re-use build artifacts.